moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -228,24 +227,26 @@ void RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::msg::RobotTrajectory& t
   if (!onedof.empty())
   {
     trajectory.joint_trajectory.header.frame_id = robot_model_->getModelFrame();
-    trajectory.joint_trajectory.header.stamp = ros::Time(0);
+    trajectory.joint_trajectory.header.stamp = stamp;
     trajectory.joint_trajectory.points.resize(waypoints_.size());
   }
 
   if (!mdof.empty())
   {
     trajectory.multi_dof_joint_trajectory.header.frame_id = robot_model_->getModelFrame();
-    trajectory.multi_dof_joint_trajectory.header.stamp = ros::Time(0);
+    trajectory.multi_dof_joint_trajectory.header.stamp = stamp;
     trajectory.multi_dof_joint_trajectory.points.resize(waypoints_.size());
   }
 
-  static const ros::Duration ZERO_DURATION(0.0);
+  static const rclcpp::Duration ZERO_DURATION(0.0);
   double total_time = 0.0;
+  rclcpp::Duration dur_total(0, 0);
   for (std::size_t i = 0; i < waypoints_.size(); ++i)
   {
     if (duration_from_previous_.size() > i)
       total_time += duration_from_previous_[i];
-
+      int seconds = (int)total_time;
+      dur_total = rclcpp::Duration((int32_t)seconds, (int32_t)((total_time-seconds)*1.0e+9));
```

Comment:
Would it be better to define a method for converting from a `double` to `rclcpp::Duration`? This logic takes a minute to process in context.

Reply:
One option would be to pass in the total nanoseconds as a `rcl_duration_value_t` which is really just an `int64_t`.

Reply:
Can you help with this @mlautman?

Reply:
I believe that all you would need is:
```cpp
#include "rcl/time.h"
rclcpp::Duration time(static_cast<rcl_duration_value_t>time_in_nanoseconds);
```
Although given that `rcl_duration_value_t` is an `int64_t` 
```cpp
# From rcl/time.h and rcutils/time.h
typedef int64_t rcutils_duration_value_t;
typedef rcutils_duration_value_t rcl_duration_value_t;
```
you should be able to simply do:
```cpp
rclcpp::Duration time(static_cast<int64_t>time_in_nanoseconds);
```

Reply:
Let me know if that doesn't work

Reply:
Hello @mlautman, tried your recommendations but failed miserably and didn't manage to get it compiling. Maybe I simply didn't understood what you meant?

Can you please make a code suggestion that compiles?